mkdocs/build.py

```diff
@@ -164,30 +166,16 @@ def build_template(template_name, env, config, site_navigation=None):
 
 def _build_page(page, config, site_navigation, env, dump_json):
 
-    # Read the input file
-    input_path = os.path.join(config['docs_dir'], page.input_path)
-
-    try:
-        input_content = io.open(input_path, 'r', encoding='utf-8').read()
-    except IOError:
-        log.error('file not found: %s', input_path)
-        raise
-
-    # Process the markdown text
-    html_content, table_of_contents, meta = convert_markdown(
-        markdown_source=input_content,
-        config=config,
-        site_navigation=site_navigation
-    )
+    html_content, toc = convert_markdown(page.markdown, config, site_navigation)
 
     context = get_global_context(site_navigation, config)
     context.update(get_page_context(
-        page, html_content, table_of_contents, meta, config
+        page, html_content, toc, page.meta, config
```

Comment:
Yeah, I agree. In my previous version I had added it to the template, but left it available via the old names too to avoid updating all the templates at the same time.